### DISCUSSION TITLE
Invalid Regular Expression Error on iPhone When Requesting Conversation

### DISCUSSION THREAD

**Discussion by gkingt** [open-webui/open-webui#3307] [General]
Bug Report
Description
Bug Summary:
When using the application on an iPhone, an error occurs: "Invalid regular expression: invalid group specifier name."

Steps to Reproduce:

Open the application on an iPhone.
Request a conversation.
Observe the error message "Invalid regular expression: invalid group specifier name."
Expected Behavior:
The application should function without errors and handle regular expressions correctly.

Actual Behavior:
An error message "Invalid regular expression: invalid group specifier name" is displayed, preventing proper functionality.

Environment
Open WebUI Version: 0.3.5

Operating System: iOS (version unknown, please specify)

Browser (if applicable): Safari (version unknown, please specify)

Reproduction Details
Confirmation:

 I have read and followed all the instructions provided in the README.md.
 I am on the latest version of both Open WebUI and Ollama.
 I have included the browser console logs.
 I have included the Docker container logs.
Logs and Screenshots
Browser Console Logs:
[Include relevant browser console logs, if applicable]

Docker Container Logs:
[Include relevant Docker container logs, if applicable]

Screenshots (if applicable):
[Attach any relevant screenshots to help illustrate the issue]

Installation Method
Manual installation

Additional Information
This error seems specific to the regular expression handling on iOS Safari. Including any error messages or stack traces might help diagnose the issue further. The error might be related to how Safari handles regular expressions differently compared to other browsers.

Note
If the bug report is incomplete or does not follow the provided instructions, it may not be addressed. Please ensure that you have followed the steps outlined in the README.md and troubleshooting.md documents, and provide all necessary information for us to reproduce and address the issue. Thank you!

**Comment by krinlove99** [comment]
I also encountered this problem

**Comment by krinlove99** [comment]
This problem is also encountered in other browsers such as Chrome-iOS

**Comment by PLK2** [comment]
Same issue occurs on an iphone 12, both Safari and Chrome, even with Open WebUI 0.3.8 installed.

**Comment by justinh-rahb** [comment]
Related:
- #3372

**Comment by kylespencer** [comment]
Running into this issue with iPhone users on Safari & Chrome & Firefox.

**Comment by krinlove99** [comment]
https://github.com/open-webui/open-webui/discussions/3372#discussioncomment-9889979

You can refer to this comment and need to modify the mentioned code because a certain version of the library for iOS.
I have confirmed that this solution works.
Reference: https://stackoverflow.com/questions/11761563/javascript-regexp-for-splitting-text-into-sentences-and-keeping-the-delimiter
